### PR TITLE
Automatically withdraw license fees to owner address

### PR DESCRIPTION
**Pull request by DeusExM4ch1n4**
This change addresses a bug found involving license fee payments getting trapped in PFA contracts. Now these fees should be paid automatically to PFA owners.

closes #1471

### CODE REVIEW

**Comment by DeusExM4ch1n4** [comment]
Adding unit test

File: contracts/PFA.sol [hunk 1]
```diff
@@ -52,13 +52,9 @@ abstract contract PFA is IPFA, LimitedOwnable {
...
     /// @notice Sets the price per access in wei for content backed
     /// by this contract.
-    function setPricePerAccess(uint256 pricePerAccess_)
-        public
-        override
-        nonReentrant
-        onlyOwner
-        afterInit
-    {
+    function setPricePerAccess(
```
Comment (un7c0rn) [inline contracts/PFA.sol]: @DeusExM4ch1n4 Can you identify what's causing the formatting issue and restore? For audit purposes / internal consistency would like to keep only the critical diffs.

File: contracts/PFA.sol [hunk 2]
```diff
@@ -144,5 +133,6 @@ abstract contract PFA is IPFA, LimitedOwnable {
         _licenseTimestamps[recipient_] = block.timestamp;
         emit License(recipient_);
         _transactionCount++;
+        payable(owner()).transfer(msg.value);
```
Comment (un7c0rn) [inline contracts/PFA.sol:147]: Can you provide detail on what the explicit cost increase will be here when adding this to the contract? We should assess whether we'll need to increase the cost estimation at playlist upload time when calling the license function. This should be small, but we need to be precise.

File: contracts/test/contract.PFA.test.js [hunk 3]
```diff
@@ -76,6 +77,55 @@ contract("PFAUnit", (accounts) => {
     }
   );
 
+  specify("Licensing fee sent to owner", async () => {
```
Comment (un7c0rn) [inline contracts/test/contract.PFA.test.js:80]: Looks great.

**Review by un7c0rn — CHANGES_REQUESTED** [review]
Thanks @DeusExM4ch1n4! Looks good, notes inline.

**Comment by un7c0rn** [comment]
@DeusExM4ch1n4 Please link issue in PR description as well.

File: contracts/PFA.sol (hunk 1 above)
Comment (DeusExM4ch1n4) [reply contracts/PFA.sol]: This change comes from the standard Prettier Solidity formatter
https://github.com/prettier-solidity/prettier-plugin-solidity

Comment (DeusExM4ch1n4) [reply contracts/PFA.sol]: I will revert the modifications to the stylization.

File: contracts/PFA.sol (hunk 2 above)
Comment (DeusExM4ch1n4) [reply contracts/PFA.sol:147]: We dynamically evaluate the cost of the deployment via the abi, so I don't think we would need to make any hardcoded changes as any change in the contract would be reflected in the compiled ABI and then in the total cost.

Comment (DeusExM4ch1n4) [reply contracts/PFA.sol:147]: https://github.com/formless-eng/share/blob/bcd850a4d3eced0876b8159b154885fdf09d686a/backend/server.js#L691

Here if you follow the definition of calculateGNFTDeploy cost you will find the dynamic estimation of gas cost.

**Review by un7c0rn — APPROVED** [review]
LGTM.

**Pull request merged**